### PR TITLE
Geoipresolver frontend upgrade

### PR DESCRIPTION
**Pull request by mikedklein**
Update GeoIpResolverConfig to our codebase standards of typescript, functional components, and Formik.

## Description
- Convert GeoIpResolverConfig to functional component
- Convert GeoIpResolverConfig to TypeScript
- Convert GeoIpResolverConfig modal to Formik Form
- EsLint Fixes
- Adjust names of functions

## Motivation and Context
We adjusted the functionality of the GeoIpConfig and made functional changes to the frontend in keeping with code stewardship we are upgrading the component to our current standards.

## How Has This Been Tested?
- Tested locally
- Unit Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by roberto-graylog — APPROVED** [review]
Functionally it looks good to me. 

The only considerations I would propose are changing the *DatabaseType nomenclature to *VendorType to make it consistent with the back-end (in the back-end we have both database types and vendor types which mean different things.

Also, for the setting on whether to enforce the Graylog Schema, in the selection display we have **Default Graylog Schema** and in the dialog/form we have **Enforce default Graylog schema**.  This may be more clear if both have the same label (Enforce default Graylog schema).

**Review by linuspahl — APPROVED** [review]
The code looks good to me, well done!

File: graylog2-web-interface/src/components/maps/configurations/GeoIpResolverConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    setCurConfig({ ...config });
+  }, [config]);
+
+  const resetConfig = () => {
+    setCurConfig(config);
+    setShowModal(false);
+  };
+
+  const handleSubmit = (values) => {
+    updateConfig(values)
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/maps/configurations/GeoIpResolverConfig.tsx:71]: I have only one optional improvement in mind. When we return the promise here, we can use Formiks loading state, to disable the save button while the request is not finished.

**Comment by mikedklein** [comment]
@linuspahl @roberto-graylog I made the suggested changes if you want to give it one more check over then I think we can merge.

File: graylog2-web-interface/src/components/maps/configurations/GeoIpResolverConfig.tsx (hunk 1 above)
Comment (linuspahl) [reply graylog2-web-interface/src/components/maps/configurations/GeoIpResolverConfig.tsx:70]: There is no need to call `setSubmitting` manually, you can also just return the promise (`updateConfig `).

**Review by roberto-graylog — APPROVED** [review]

**Review by danotorrey — APPROVED** [review]
LGTM and tests successfully

**Comment by danotorrey** [comment]
@mikedklein Feel free to merge when green

**Pull request merged**